### ISSUE TITLE
Missing XGBoostRanker in xgboost4j-spark jvm package

### ISSUE THREAD

**Issue by Meleeri** [dmlc/xgboost#10118]
On the latest version (2.03) of xgboost4j-spark XGBoostClassifer and XGBoostRegressor are available for classification and regression trainning tasks, but there seems to be no equivalent pipeline for learning-to-rank tasks. How should I train a ranking model using current pipelines or lower level apis？Has anyone successfully run it on spark before？

**Comment by wbo4958** [comment]
No, XGBoost JVM package doesn't support XGBoostRank for now, but I think it's the right time to support it in XGBoost. @trivialfis

**Comment by trivialfis** [comment]
The JVM package implements LTR as regression. Yeah, we can have some additional features in there, but let me finish the networking first, it's a huge time killer

**Comment by wbo4958** [comment]
Yeah, We should move this part from regression to XGBoostRank just like xgboost pyspark
